bundles/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/handler/GardenaAccountHandler.java

```diff
@@ -51,80 +62,191 @@
 @NonNullByDefault
 public class GardenaAccountHandler extends BaseBridgeHandler implements GardenaSmartEventListener {
     private final Logger logger = LoggerFactory.getLogger(GardenaAccountHandler.class);
-    private static final long REINITIALIZE_DELAY_SECONDS = 120;
-    private static final long REINITIALIZE_DELAY_HOURS_LIMIT_EXCEEDED = 24;
 
-    private @Nullable GardenaDeviceDiscoveryService discoveryService;
+    // timing constants
+    private static final Duration REINITIALIZE_DELAY_SECONDS = Duration.ofSeconds(120);
+    private static final Duration REINITIALIZE_DELAY_MINUTES_BACK_OFF = Duration.ofMinutes(15).plusSeconds(30);
+    private static final Duration REINITIALIZE_DELAY_HOURS_LIMIT_EXCEEDED = Duration.ofHours(24).plusMinutes(15);
+
+    // localisation constants
+    private static final String RECONNECT_MSG_KEY = "accounthandler.waiting-to-reconnect-at-time";
+    private static final DateTimeFormatter DATE_TIME_FORMAT = DateTimeFormatter.ofPattern(" HH:MM:SS (cccc)");
 
+    // assets
+    private @Nullable GardenaDeviceDiscoveryService discoveryService;
     private @Nullable GardenaSmart gardenaSmart;
-    private HttpClientFactory httpClientFactory;
-    private WebSocketFactory webSocketFactory;
+    private final HttpClientFactory httpClientFactory;
+    private final WebSocketFactory webSocketFactory;
+    private final Bundle bundle;
+    private final TranslationProvider i18nProvider;
+    private final LocaleProvider localeProvider;
 
-    public GardenaAccountHandler(Bridge bridge, HttpClientFactory httpClientFactory,
-            WebSocketFactory webSocketFactory) {
+    // re- initialisation stuff
+    private final Object reInitializationCodeLock = new Object();
+    private @Nullable ScheduledFuture<?> reInitializationTask;
+    private boolean reInitializationCausedBy429 = false;
+    private Instant lastApiCallTime = Instant.MIN;
+    private boolean lastApiCallTimeLoaded = false;
+
+    public GardenaAccountHandler(Bridge bridge, HttpClientFactory httpClientFactory, WebSocketFactory webSocketFactory,
+            Bundle bundle, TranslationProvider i18nProvider, LocaleProvider localeProvider) {
         super(bridge);
         this.httpClientFactory = httpClientFactory;
         this.webSocketFactory = webSocketFactory;
+        this.bundle = bundle;
+        this.i18nProvider = i18nProvider;
+        this.localeProvider = localeProvider;
+    }
+
+    /**
+     * Returns the time when the last api call was made. If the thing was newly created, restores the time from a
+     * property to ensure consistent behaviour over restarts.
+     *
+     * @return the time when the last api call was made.
+     */
+    private Instant lastApiCallTime() {
```

Comment:
Perhaps `getLastApiCallTime()` would be more descriptive?

Reply:
Api call suppression concept modified.